### PR TITLE
Add pre-make helper for checking dependencies

### LINKED ISSUE #27066: Add helper for checking dependencies on make
### Motivation

It's a common issue where the versions of dependencies become out of sync with the required versions for the version of moose.

### Design

Add a helper script that will warn when these dependencies are out of date, when possible.

### Impact

Will make it clearer when things are out of date instead of relying on understanding errors.

### PR DESCRIPTION
**Pull request by loganharbour**
Closes #27066

### CODE REVIEW

**Comment by loganharbour** [comment]
Still need to add a link to suggest you visit when things are out of date

File: framework/moose.mk [hunk 1]
```diff
@@ -376,6 +376,12 @@ endif
 wasp_submodule_status:
 	@if [ x$(wasp_submodule_message) != "x" ]; then printf $(wasp_submodule_message); exit 1; fi
 
+# pre-make for checking current dependency versions and showing useful warnings
+# if things like conda packages are out of date
+prebuild:
+	@-python $(FRAMEWORK_DIR)/../scripts/premake.py
```
Comment (dschwen) [inline framework/moose.mk:382]:
```suggestion
	@-python3 $(FRAMEWORK_DIR)/../scripts/premake.py
```

?

Comment (milljm) [reply framework/moose.mk:382]: Something like this maybe: https://github.com/idaholab/moose/blob/2fd248e338982ab5b2dc53548af01210c5aa7ace/framework/moose.mk#L113

**Comment by loganharbour** [comment]
@dschwen it would be good to have your input on the new `prebuild` target. @permcody and I couldn't think of a better way to do it, and this will unfortunately require adding `prebuild` to other custom targets in apps, which kinda sucks...

**Comment by dschwen** [comment]
How about https://www.gnu.org/software/make/manual/html_node/Parallel-Disable.html
...
We might not be able to specify an order-only dependency _before_ all regular dependencies though...

**Review by permcody — APPROVED** [review]
This looks good to me. Make is at least half art. We can always pull back the dependencies if we have problems.

**Comment by loganharbour** [comment]
@dschwen I'm open to other suggestions, but I haven't gotten any of yours to work. 

I think that this is pretty important for user support and am willing to do it the _better_ way, but I can't find one.

**Review by milljm — APPROVED** [review]
💯

**Review by dschwen — APPROVED** [review]

**Comment by loganharbour** [comment]
Coverage failures are unrelated.

**Pull request merged**